parsl/app/bash.py

```diff
@@ -123,6 +124,13 @@ def __init__(self, func, data_flow_kernel=None, cache=False, executors='all', ig
             if sig.parameters[s].default is not Parameter.empty:
                 self.kwargs[s] = sig.parameters[s].default
 
+        # update_wrapper allows remote_side_bash_executor to masquerade as self.func
+        # partial is used to attach the first arg the "func" to the remote_side_bash_executor
+        # this is done to avoid passing a function type in the args which parsl.serializer
+        # doesn't support
```

Comment:
I'm using functions as args (to bash apps) in the LSST DRP workflow. I think you would have to argue hard with me to convince me it is better to prohibit that and for me to move to a different style.

Reply:
I was worried about this. If the function can be serialized by pickle, this shouldn't break. Can you point me to your code ?

Reply:
With the updates, the serializer will try pickle and if that fails - dill to try serialize your params. From testing pickle and dill should be able to handle most situations.

Reply:
so this comment should be more that function types are supported less well?